server/task/task-distributed/src/main/java/org/apache/james/task/eventsourcing/distributed/RabbitMQWorkQueue.java

```diff
@@ -109,7 +109,7 @@ private void consumeWorkqueue() {
         receiver = new Receiver(new ReceiverOptions().connectionMono(connectionMono));
         receiverHandle = receiver.consumeManualAck(QUEUE_NAME, new ConsumeOptions())
             .subscribeOn(Schedulers.boundedElastic())
-            .flatMap(this::executeTask)
+            .concatMap(this::executeTask)
```

Comment:
I'm always afraid to see such important change without a test to check it. It's so easy to break a behavior like that, we should ensure a test prevents it.

Reply:
The fact that the following test is stable is due to this change. 
I have a specific test in mind to demonstrate and enforce this change.